### PR TITLE
Add CRUD users tests.

### PR DESCRIPTION
**Pull request by kersommoura**
Add CRUD user tests for `Pulp3`.

See: #748

### CODE REVIEW

File: pulp_smash/tests/pulp3/pulpcore/api_v3/test_crud_repos.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def test_02_read_repo(self):
     def test_02_read_repos(self):
         """Search for the repository by its name.
 
-        Assert that just one search result is returned, and that that result
+        Assert that just one search result is returned, and that result
```
Comment (Ichimonji10) [inline pulp_smash/tests/pulp3/pulpcore/api_v3/test_crud_repos.py]: This is a nice grammar improvement, but it's not related to the rest of this test. While we're at it, it'd be good to change the line wrapping:

    Assert that just one search result is returned, and that result has a
    correct name.

Comment (kersommoura) [reply pulp_smash/tests/pulp3/pulpcore/api_v3/test_crud_repos.py]: @Ichimonji10 , done. If there is any minor update in the future will be in a separate PR.

File: pulp_smash/tests/pulp3/pulpcore/api_v3/test_crud_repos.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def test_02_read_repo(self):
     def test_02_read_repos(self):
         """Search for the repository by its name.
 
-        Assert that just one search result is returned, and that that result
-        has a correct name.
+        Assert that just one search result is returned, and that result has a
```
Comment (Ichimonji10) [inline pulp_smash/tests/pulp3/pulpcore/api_v3/test_crud_repos.py]: The point I was driving at in my previous comment is that this change shouldn't be made as part of this PR. This change is logically unrelated from the rest of this PR, and should be submitted as a separate PR.

Speaking of grammar, this change actually makes less grammatical sense. If you'd like to change this sentence to be less redundant, use a construct like "that the result has a correct name," not "that result has a correct name."

**Comment by Ichimonji10** [comment]
Merged with a follow-up commit of c9fa2f7a02aea59e0dfc123f079e5dde1ac95da8.

**Pull request merged**